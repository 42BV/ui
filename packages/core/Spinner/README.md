# @42.nl/ui-core-spinner

## Installation

Install the package in your project directory with:

```sh
// with npm
npm install @42.nl/ui-core-spinner

// with yarn
yarn add @42.nl/ui-core-spinner
```
