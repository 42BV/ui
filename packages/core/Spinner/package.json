{
  "name": "@42.nl/ui-core-spinner",
  "version": "0.0.1",
  "description": "",
  "main": "lib/Spinner.js",
  "private": false,
  "scripts": {
    "test": "jest --config=jest.config.js",
    "build": "tsc",
    "lint": "eslint src --ext js,jsx,ts,tsx --max-warnings=0",
    "clean": "rm -rf lib"
  },
  "files": [
    "lib/*"
  ],
  "dependencies": {
    "classnames": "2.2.6"
  },
  "peerDependencies": {
    "react": "^16.9.0",
    "reactstrap": "^8.0.1"
  },
  "devDependencies": {
    "react": "16.9.0",
    "reactstrap": "8.0.1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "publishConfig": {
    "access": "public"
  }
}
