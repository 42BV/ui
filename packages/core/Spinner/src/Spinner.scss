// @remove-on-build-start
@import '~@42.nl/ui-core-styling/lib/bootstrap';
// @remove-on-build-end

.spinner {
  animation: rotate 2s linear infinite;

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  .path {
    transform-origin: center;
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
    animation: dash 1.5s ease-in-out infinite;
    stroke-linecap: round;
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 200;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 89, 200;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 89, 200;
      stroke-dashoffset: -124;
    }
  }
}
