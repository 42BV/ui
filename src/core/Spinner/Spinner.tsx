import React from 'react';
import classNames from 'classnames';

type Props = {
  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;

  /**
   * Color of the circle within the spinner.
   *
   */
  color: string;

  /**
   * Width and height of the spinner.
   *
   */
  size: number;
};

/**
 * Spinner is a component which is used as a loading indicator.
 *
 * Use it for example when data is fetching during navigation to a page.
 */
export function Spinner({ className, color, size }: Props) {
  return (
    <svg
      className={classNames('spinner', className)}
      width={size}
      height={size}
      viewBox="25 25 50 50"
    >
      <circle
        className="path"
        fill="none"
        stroke={color}
        cx="50"
        cy="50"
        r="20"
        strokeWidth="4"
        strokeMiterlimit="10"
      />
    </svg>
  );
}
