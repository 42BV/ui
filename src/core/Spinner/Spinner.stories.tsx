import React from 'react';

import { Spinner } from './Spinner';

export default {
  title: 'core/Spinner',

  parameters: {
    component: Spinner
  }
};

const DefaultStory = () => {
  return (
    <div className="text-center">
      <Spinner color="" size={42} />
      <hr />
      <Spinner color="red" size={42} />
      <Spinner color="white" size={42} />
      <Spinner color="blue" size={42} />
      <hr />
      <Spinner color="orange" size={16} />
      <hr />
      <Spinner color="purple" size={500} />
    </div>
  );
};

export const Default = {
  render: DefaultStory,
  name: 'default'
};
