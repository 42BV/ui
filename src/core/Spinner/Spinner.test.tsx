import React from 'react';
import { render } from '@testing-library/react';

import { Spinner } from './Spinner';

describe('Component: Spinner', () => {
  test('ui', () => {
    const { container } = render(
      <Spinner color="white" size={42} />
    );
    expect(container).toMatchSnapshot();
  });
});
