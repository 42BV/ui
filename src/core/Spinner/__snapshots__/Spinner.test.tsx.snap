// Vitest Snapshot v1

exports[`Component: Spinner > ui 1`] = `
<div>
  <svg
    class="spinner"
    height="42"
    viewBox="25 25 50 50"
    width="42"
  >
    <circle
      class="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      stroke="white"
      stroke-miterlimit="10"
      stroke-width="4"
    />
  </svg>
</div>
`;
